hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/controller/ContractControllerTest.java

```diff
@@ -0,0 +1,167 @@
+package com.hedera.mirror.web3.controller;
+
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.NOT_IMPLEMENTED;
+
+import javax.annotation.Resource;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import org.springframework.web.reactive.function.BodyInserters;
+
+@ExtendWith(SpringExtension.class)
+@WebFluxTest(controllers = ContractController.class)
+class ContractControllerTest {
+    private static final String CALL_URI = "/api/v1/contracts/call";
+    private static final String NOT_IMPLEMENTET_ERROR = "Operations eth_call and gas_estimate are not supported yet!";
+
+    @Resource
+    private WebTestClient webClient;
+
+    @Test
+    void throwsUnsupportedOperationExceptionWhenCalled() {
+
+        webClient.post()
+                .uri(CALL_URI)
+                .contentType(MediaType.APPLICATION_JSON)
+                .body(BodyInserters.fromValue(request()))
+                .exchange()
+                .expectStatus()
+                .isEqualTo(NOT_IMPLEMENTED)
+                .expectBody(GenericErrorResponse.class)
+                .isEqualTo(new GenericErrorResponse(NOT_IMPLEMENTET_ERROR));
+    }
+
+    @Test
+    void throwsValidationExceptionWhenCalledWithMissingToField() {
+        final var request = request();
+        request.setTo("");
```

Comment:
We should use parameterized tests to quickly test many different variations of input. Test null, empty string, blanks, non-hex chars, longer than 40 chars, etc. Same for other invalid field tests.

```suggestion
    @ValueSource(strings = {"null", "", " ", "0x", "0xghijklmno...", "0x00...ad"})
    @ParameterizedTest
    void callWhenInvalidTo(String to) {
        final var errorString = AddressValidator.MESSAGE;
        final var request = request();
        request.setTo(to);
```